src/bwc_staking_contract.cairo

```diff
@@ -87,26 +78,18 @@ mod BWCStakingContract {
 
     #[external(v0)]
     impl StakingTokenTraitImpl of super::StakingTokenTrait<ContractState> {
-        // fn depositBWC(ref self: ContractState, amount: u256) -> bool {
-        //     IBWCERC20TokenDispatcher { contract_address: BWCERC20TokenAddr }
-        //         .transfer_from(
-        //             contract_address = reward_token_address,
-        //             sender = caller_address,
-        //             recipient = contract_address,
-        //             amount: u256
-        //         )
-        // }
         fn stake_bwc_token(
             ref self: ContractState, amount: u256, BWCERC20TokenAddr: ContractAddress
         ) -> bool {
             let caller: ContractAddress = get_caller_address();
             let address_this = get_contract_address();
```

Comment:
add validation logic to ensure that only the contract address of `BWC` token can be used to stake